resources/mediaplayer/src/main/java/org/mobicents/media/server/impl/resource/mediaplayer/audio/wav/WavTrackImpl.java

```diff
@@ -35,104 +41,65 @@
     private final static byte ULAW_PADDING_BYTE = (byte) 0xFF;
 
     private final static byte[] factBytes = new byte[] { 0x66, 0x61, 0x63, 0x74 };
     private byte paddingByte = PCM_PADDING_BYTE;
-
-    public WavTrackImpl(URL url) throws UnsupportedAudioFileException, IOException {
-        inStream = url.openStream();
-
-        getFormat(inStream);
+    private static final Logger logger = Logger.getLogger(WavTrackImpl.class);
+    
+    public WavTrackImpl (URL url) throws IOException, UnsupportedAudioFileException {
+        
+        buff = ByteBuffer.allocate(BUFFER_SIZE); 
+        rbChannel=Channels.newChannel(url.openStream());
+        
+        getAudioFormat(rbChannel);
         if (format == null) {
             throw new UnsupportedAudioFileException();
         }
+       
     }
 
-    public void setPeriod(int period) {
-        this.period = period;
-        frameSize = (int) (period * format.getChannels() * format.getSampleSize() * format.getSampleRate() / 8000);
-    }
-
-    public int getPeriod() {
-        return period;
-    }
-
-    @Override
-    public long getMediaTime() {
-        return 0;// timestamp * 1000000L;
-    }
-
-    @Override
-    public long getDuration() {
-        return duration;
-    }
-
-    @Override
-    public void setMediaTime(long timestamp) {
-        // this.timestamp = timestamp/1000000L;
-        // try {
-        // long offset = frameSize * (timestamp / period);
-        // byte[] skip = new byte[(int)offset];
-        // stream.read(skip);
-        // } catch (IOException e) {
-        // }
-    }
-
-    private void skip(long timestamp) {
-        try {
-            long offset = frameSize * (timestamp / period / 1000000L);
-            byte[] skip = new byte[(int) offset];
-            int bytesRead = 0;
-            while (bytesRead < skip.length && inStream.available() > 0) {
-                int len = inStream.read(skip, bytesRead, skip.length - bytesRead);
-                if (len == -1)
-                    return;
-
-                totalRead += len;
-                bytesRead += len;
-            }
-
-        } catch (IOException e) {
-            logger.error(e);
-        }
-    }
-
-    private void getFormat(InputStream stream) throws IOException {
+    private void getAudioFormat(ReadableByteChannel rbc) throws IOException {
         byte[] header = new byte[36];
         byte[] headerEnd = null;
+       
+        int bytesReadFromChannel = 0 ;
         int bytesRead = 0;
-        while (bytesRead < 36 && stream.available() > 0) {
-            int len = stream.read(header, bytesRead, 36 - bytesRead);
-            if (len == -1) {
-                return;
-            }
-            bytesRead += len;
+        int count = 0;
+            
+        bytesReadFromChannel=rbc.read(buff);
+        buff.clear();
+        
+        while (count < 36 && buff.remaining() > 0) {
+            //count=rbc.read(buffer);
+            header[count]=buff.get(count);
```

Comment:
can we use bulk here?https://docs.oracle.com/javase/7/docs/api/java/nio/ByteBuffer.html#get(byte[],%20int,%20int)